autogluon/utils/tabular/contrib/tabular_nn_pytorch/nn_tab_model.py

```diff
@@ -0,0 +1,314 @@
+            objective_func_name = self.params['nn.tabular.metric']
+            nn_metric = self.metrics_map[self.params['nn.tabular.metric']]
+        else:
+            nn_metric = None
+        return nn_metric, objective_func_name
+
+    def __get_objective_func_to_monitor(self, objective_func_name):
+        monitor_obj_func = {
+            'roc_auc': 'auroc',
+
+            'f1': 'f_beta',
+            'f1_macro': 'f_beta',
+            'f1_micro': 'f_beta',
+            'f1_weighted': 'f_beta',
+
+            'precision_macro': 'precision',
+            'precision_micro': 'precision',
+            'precision_weigthed': 'precision',
+
+            'recall_macro': 'recall',
+            'recall_micro': 'recall',
+            'recall_weigthed': 'recall',
+            'log_loss': 'valid_loss',
+        }
+        objective_func_name_to_monitor = objective_func_name
+        if objective_func_name in monitor_obj_func:
+            objective_func_name_to_monitor = monitor_obj_func[objective_func_name]
+        return objective_func_name_to_monitor
+
+    def predict(self, X, preprocess=True):
+        return super().predict(X, preprocess)
+
+    def predict_proba(self, X, preprocess=True):
+        if preprocess:
+            X = self.preprocess(X)
+        self.model.data.add_test(TabularList.from_df(X, cat_names=self.cat_names, cont_names=self.cont_names, procs=self.procs))
+        with progress_disabled_ctx(self.model) as model:
+            preds, _ = model.get_preds(ds_type=DatasetType.Test)
+        if self.problem_type == REGRESSION:
+            if self.y_scaler is not None:
+                return self.y_scaler.inverse_transform(preds.numpy()).reshape(-1)
+            else:
+                return preds.numpy().reshape(-1)
+        if self.problem_type == BINARY:
+            return preds[:, 1].numpy()
+        else:
+            return preds.numpy()
+
+    def save(self, file_prefix='', directory=None, return_filename=False, verbose=True):
+        # Export model
+        save_pkl.save_with_fn(f'{self.path}{self.model_internals_file_name}', self.model, lambda m, buffer: m.export(buffer, destroy=True), verbose=verbose)
+        self.model = {}
+        return super().save(file_prefix=file_prefix, directory=directory, return_filename=return_filename, verbose=verbose)
+
+    @classmethod
+    def load(cls, path: str, file_prefix='', reset_paths=False, verbose=True):
+        obj = super().load(path, file_prefix=file_prefix, reset_paths=reset_paths, verbose=verbose)
+        obj.model = load_pkl.load_with_fn(f'{obj.path}{obj.model_internals_file_name}', lambda p: load_learner(obj.path, p), verbose=verbose)
+        return obj
+
+    def _get_default_searchspace(self, problem_type):
+        return get_default_searchspace(problem_type)
+
+    def hyperparameter_tune(self, X_train, X_test, Y_train, Y_test, scheduler_options=None, **kwargs):
+        self.fit(X_train=X_train, X_test=X_test, Y_train=Y_train, Y_test=Y_test, **kwargs)
```

Comment:
Add TODO that hyperparameter-tuning is not yet implemented.  I think however it would be good to add HPO before merging the PR, since many users may want to use HPO if they're using this non-default model.